crates/bevy_asset/src/server/mod.rs

```diff
@@ -1716,24 +1714,33 @@ pub fn handle_internal_asset_events(world: &mut World) {
         };
 
         let mut paths_to_reload = <HashSet<_>>::default();
+        let mut reload_path = |path: PathBuf, source: &AssetSourceId<'static>| {
+            let path = AssetPath::from(path).with_source(source);
+            if infos.should_reload(&path) {
```

Comment:
I don't think this should_reload should be here? If you have an asset loader that loads a nested asset in "immediate" mode, I don't think a handle will be kept alive, but the loader_dependents will be updated, and we should trigger a reload for that root loader (since maybe previously its nested asset was missing and now its not missing.

Actually looks like `NestedLoader<_, _, _, Immediate>` doesn't even record the dependencies, which likely means it'll get out of sync right now. Funnnn. Regardless of that issue, I don't think we should be filtering on whether the handle is alive. The `loader_dependents` will be empty if the path is truly unused.